Python: upgrade package set

FRidh opened the pull request:
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

See https://hydra.nixos.org/jobset/nixpkgs/python-unstable

FRidh committed:
python.pkgs.pluggy: 0.3.1 -> 0.6.0

FRidh committed:
python.pkgs.six: disable tests to prevent infinite recursion

FRidh committed:
awscli: 1.14.6 -> 1.14.17

FRidh committed:
python.pkgs.texttable: 0.8.4 -> 1.1.1

FRidh committed:
python.pkgs.docker_compose: fixup

FRidh committed:
python: absl-py: 0.1.5 -> 0.1.7

FRidh committed:
python: aenum: 2.0.8 -> 2.0.9

FRidh committed:
python: aiohttp: 2.3.3 -> 2.3.7

FRidh committed:
python: APScheduler: 3.4.0 -> 3.5.0

FRidh committed:
python: arrow: 0.10.0 -> 0.12.0

FRidh committed:
python: asgiref: 1.1.2 -> 2.0.1

FRidh committed:
python: asn1crypto: 0.23.0 -> 0.24.0

FRidh committed:
python: astor: 0.5 -> 0.6.2

FRidh committed:
python: astroid: 1.5.3 -> 1.6.0

FRidh committed:
python: astropy: 2.0.2 -> 2.0.3

FRidh committed:
python: attrs: 17.2.0 -> 17.4.0

FRidh committed:
python: bokeh: 0.12.10 -> 0.12.13

FRidh committed:
python: botocore: 1.8.10 -> 1.8.21

FRidh committed:
python: credstash: 1.13.4 -> 1.14.0

FRidh committed:
python: csscompressor: 0.9.4 -> 0.9.5

FRidh committed:
python: cx_Freeze: 5.0.2 -> 5.1.1

FRidh committed:
python: cytoolz: 0.8.2 -> 0.9.0

FRidh committed:
python: dask: 0.15.4 -> 0.16.0

FRidh committed:
python: distro: 1.0.4 -> 1.2.0

FRidh committed:
python: django-jinja: 2.2.2 -> 2.4.1

FRidh committed:
python: djangorestframework: 3.7.3 -> 3.7.7

FRidh committed:
python: docker: 2.5.1 -> 2.7.0

FRidh committed:
python: docker-compose: 1.15.0 -> 1.18.0

FRidh committed:
python: easy-thumbnails: 2.4.2 -> 2.5

FRidh committed:
python: Eve: 0.7.4 -> 0.7.5

FRidh committed:
python: Faker: 0.8.7 -> 0.8.8

FRidh committed:
python: faulthandler: 2.6 -> 3.0

FRidh committed:
python: filelock: 2.0.13 -> 2.0.14

FRidh committed:
python: Fiona: 1.7.10.post1 -> 1.7.11

FRidh committed:
python: flake8-debugger: 1.4.0 -> 3.0.0

FRidh committed:
python: Flask-Testing: 0.6.2 -> 0.7.1

FRidh committed:
python: flit: 0.11.4 -> 0.13

FRidh committed:
python: fonttools: 3.17.0 -> 3.21.0

FRidh committed:
python: ftfy: 5.1.1 -> 5.2.0

FRidh committed:
python: gensim: 3.0.1 -> 3.2.0

FRidh committed:
python: google-api-core: 0.1.2 -> 0.1.3

FRidh committed:
python: GPy: 1.8.4 -> 1.8.5

FRidh committed:
python: grpcio: 1.7.3 -> 1.8.2

FRidh committed:
python: gssapi: 1.2.0 -> 1.3.0

FRidh committed:
python: htmlmin: 0.1.11 -> 0.1.12

FRidh committed:
python: httpbin: 0.5.0 -> 0.6.2

FRidh committed:
python: ipykernel: 4.6.1 -> 4.7.0

FRidh committed:
python: ipywidgets: 7.0.5 -> 7.1.0

FRidh committed:
python: jedi: 0.11.0 -> 0.11.1

FRidh committed:
python: jsbeautifier: 1.7.4 -> 1.7.5

FRidh committed:
python: jsonpatch: 1.16 -> 1.20

FRidh committed:
python: jupyter_client: 5.1.0 -> 5.2.0

FRidh committed:
python: jupyter_core: 4.3.0 -> 4.4.0

FRidh committed:
python: Keras: 2.1.1 -> 2.1.2

FRidh committed:
python: keyring: 10.4.0 -> 10.5.1

FRidh committed:
python: python-ldap: 2.4.45 -> 2.5.2

FRidh committed:
python: ldap3: 2.3 -> 2.4

FRidh committed:
python: line_profiler: 2.0 -> 2.1.2

FRidh committed:
python: llfuse: 1.0 -> 1.3.2

FRidh committed:
python: llvmlite: 0.20.0 -> 0.21.0

FRidh committed:
python: marionette-harness: 4.1.0 -> 4.3.0

FRidh committed:
python: marshmallow: 2.14.0 -> 2.15.0

FRidh committed:
python: matplotlib: 2.1.0 -> 2.1.1

FRidh committed:
python: MechanicalSoup: 0.8.0 -> 0.9.0.post4

FRidh committed:
python: mistune: 0.7.4 -> 0.8.3

FRidh committed:
python: moto: 0.4.31 -> 1.1.25

FRidh committed:
python: nbxmpp: 0.6.0 -> 0.6.2

FRidh committed:
python: Nikola: 7.8.10 -> 7.8.11

FRidh committed:
python: nilearn: 0.3.1 -> 0.4.0

FRidh committed:
python: notebook: 5.2.1 -> 5.2.2

FRidh committed:
python: numba: 0.35.0 -> 0.36.2

FRidh committed:
python: odfpy: 1.3.5 -> 1.3.6

FRidh committed:
python: pathlib2: 2.2.1 -> 2.3.0

FRidh committed:
python: path.py: 10.4 -> 10.5

FRidh committed:
python: pexpect: 4.3.0 -> 4.3.1

FRidh committed:
python: phonenumbers: 8.8.6 -> 8.8.8

FRidh committed:
python: pip-tools: 1.10.2 -> 1.11.0

FRidh committed:
python: plotly: 2.1.0 -> 2.2.3

FRidh committed:
python: plumbum: 1.6.3 -> 1.6.5

FRidh committed:
python: psutil: 5.4.1 -> 5.4.2

FRidh committed:
python: py: 1.4.34 -> 1.5.2

FRidh committed:
python: pyasn1: 0.3.4 -> 0.4.2

FRidh committed:
python: pyasn1-modules: 0.1.5 -> 0.2.1

FRidh committed:
python: pyblake2: 0.9.3 -> 1.1.0

FRidh committed:
python: PyChromecast: 0.8.1 -> 1.0.3

FRidh committed:
python: pycollada: 0.5 -> 0.6

FRidh committed:
python: pydot: 1.2.3 -> 1.2.4

FRidh committed:
python: pygit2: 0.26.0 -> 0.26.3

FRidh committed:
python: pyglet: 1.2.4 -> 1.3.0

FRidh committed:
python: pylast: 1.9.0 -> 2.0.0

FRidh committed:
python: pylint: 1.7.4 -> 1.8.1

FRidh committed:
python: pymongo: 3.5.1 -> 3.6.0

FRidh committed:
python: pyobjc: 4.0b1 -> 4.1

FRidh committed:
python: pyopencl: 2017.2 -> 2017.2.2

FRidh committed:
python: PySoundFile: 0.8.1 -> 0.9.0.post1

FRidh committed:
python: pytest: 3.2.5 -> 3.3.1

FRidh committed:
python: pytest-localserver: 0.3.7 -> 0.4.1

FRidh committed:
python: pytest-xdist: 1.20.1 -> 1.21.0

FRidh committed:
python: pywinrm: 0.2.2 -> 0.3.0

FRidh committed:
python: relatorio: 0.7.1 -> 0.8.0

FRidh committed:
python: restview: 2.7.0 -> 2.8.0

FRidh committed:
python: robomachine: 0.6 -> 0.8.0

FRidh committed:
python: ropper: 1.10.10 -> 1.11.2

FRidh committed:
python: Scrapy: 1.4.0 -> 1.5.0

FRidh committed:
python: seaborn: 0.7.1 -> 0.8.1

FRidh committed:
python: serpy: 0.2.0 -> 0.3.1

FRidh committed:
python: Shapely: 1.6.2.post1 -> 1.6.3

FRidh committed:
python: simplejson: 3.11.1 -> 3.13.2

FRidh committed:
python: smart_open: 1.5.3 -> 1.5.6

FRidh committed:
python: SQLAlchemy: 1.1.15 -> 1.2.0

FRidh committed:
python: sqlmap: 1.1.11 -> 1.1.12

FRidh committed:
python: stevedore: 1.27.1 -> 1.28.0

FRidh committed:
python: stripe: 1.70.0 -> 1.77.0

FRidh committed:
python: supervise_api: 0.1.5 -> 0.2.0

FRidh committed:
python: sybil: 1.0.5 -> 1.0.6

FRidh committed:
python: tabulate: 0.7.7 -> 0.8.2

FRidh committed:
python: textacy: 0.4.1 -> 0.5.0

FRidh committed:
python: Theano: 0.9.0 -> 1.0.1

FRidh committed:
python: thespian: 3.8.3 -> 3.9.0

FRidh committed:
python: toolz: 0.8.2 -> 0.9.0

FRidh committed:
python: tqdm: 4.19.4 -> 4.19.5

FRidh committed:
python: tzlocal: 1.4 -> 1.5.1

FRidh committed:
python: vowpalwabbit: 8.3.2 -> 8.4.0

FRidh committed:
python: websockets: 3.4 -> 4.0.1

FRidh committed:
python: Werkzeug: 0.12.2 -> 0.13

FRidh committed:
python: widgetsnbextension: 3.0.8 -> 3.1.0

FRidh committed:
python: ws4py: 0.4.2 -> 0.4.3

FRidh committed:
python: yapf: 0.19.0 -> 0.20.0

FRidh committed:
python: yarl: 0.13.0 -> 0.17.0

FRidh committed:
python.pkgs.attrs: disable tests to prevent infinite recursion

FRidh committed:
python.pkgs.structlog: move expression

FRidh committed:
python: structlog: 16.1.0 -> 17.2.0

FRidh committed:
python.pkgs.tox: 2.4.1 -> 2.9.1

FRidh committed:
python.pkgs.backports_functools_lru_cache: move expression

FRidh committed:
python: backports.functools_lru_cache: 1.3 -> 1.4

FRidh committed:
python.pkgs.extras: move expression

FRidh committed:
python: extras: 0.0.3 -> 1.0.0

FRidh committed:
python: testtools: 1.9.0 -> 2.3.0

FRidh committed:
python.pkgs.lxml: move expression

FRidh committed:
python: lxml: 3.8.0 -> 4.1.1

FRidh committed:
Python: disable user site-packages in setup hook

Wrappers already included `PYTHONNOUSERSITE=1`, but now this env var is
also set in the Python setup hook. This improves purity in case of
non-sandboxes builds and nix-shell.

FRidh committed:
python.pkgs.setuptools: 38.2.3 -> 38.2.5

FRidh committed:
python.pkgs.pyaml: move expression

FRidh committed:
python: pyaml: 15.02.1 -> 17.12.1

FRidh committed:
python.pkgs.aws-xray-sdk: init at 0.95

FRidh committed:
python.pkgs.boto3: move expression

FRidh committed:
python.pkgs.jsondiff: init at 1.1.1

FRidh committed:
python.pkgs.jsonpickle: init at 0.9.5

FRidh committed:
python.pkgs.pytoml: 0.1.11 -> 0.1.14

FRidh committed:
python: jsonpatch: 1.20 -> 1.21

FRidh committed:
python.pkgs.idna: move expression

FRidh committed:
python: idna: 2.5 -> 2.6

FRidh committed:
python.pkgs.pyglet: fix build

FRidh committed:
python.pkgs.libusb1: fix build

FRidh committed:
python.pkgs.ropper: fix build

FRidh committed:
python.pkgs.pytools: fix expression

FRidh committed:
python.pkgs.pytools: 2017.4 -> 2017.6

FRidh committed:
python.pkgs.pyopencl: fix expression

FRidh committed:
python.pkgs.markdown: 2.6.8 -> 2.6.10

FRidh committed:
python.pkgs.Mako: move expression

FRidh committed:
python: Mako: 1.0.4 -> 1.0.7

FRidh committed:
python.pkgs.iso8601: move expression

FRidh committed:
python: iso8601: 0.1.11 -> 0.1.12

FRidh committed:
python.pkgs.html5lib: move expression

FRidh committed:
python: html5lib: 0.999999999 -> 1.0.1

FRidh committed:
python.pkgs.kombu_3: remove expr

- two versions available
- this is the older version
- unused

FRidh committed:
python.pkgs.amqp_1: remove expr

- two versions available
- this is the older version
- unused

FRidh committed:
python.pkgs.backports_abc: move expression

FRidh committed:
python: backports_abc: 0.4 -> 0.5

FRidh committed:
python.pkgs.backports_lzma: move expression

FRidh committed:
python: backports.lzma: 0.0.3 -> 0.0.8

FRidh committed:
python.pkgs.biopython: move expression

FRidh committed:
python: biopython: 1.68 -> 1.70

FRidh committed:
python: cached-property: 1.3.0 -> 1.3.1

FRidh committed:
python.pkgs.decorator: move expression

globin commented on pkgs/development/python-modules/html5lib/default.nix:
```diff
@@ -11,11 +11,11 @@
 buildPythonPackage rec {
   pname = "html5lib";
-  version = "0.999999999";
+  version = "1.0.1";
```
:tada: they've managed it

FRidh committed:
python.pkgs.ldap3: remove name attr

FRidh committed:
mitmproxy: fix build

FRidh committed:
haxor-news: 0.4.2 -> 0.4.3

For some reason tests are failing. Note they were already failing before
this update.

FRidh committed:
python.pkgs.jdcal: 1.0 -> 1.3

FRidh committed:
python.pkgs.zope_copy: move expr

FRidh committed:
watson: 1.4.0 -> 1.5.2, fix build with patch

FRidh committed:
python.pkgs.xarray: 0.9.6 -> 0.10.0
